### PR TITLE
Added ReflectionAttribute related Documentation.

### PR DESCRIPTION
**Pull request by mumumu**
Documented `ReflectionAttribute` based on RFC and implementation.

- https://wiki.php.net/rfc/attributes_v2
- https://github.com/php/php-src/blob/php-8.0.0/ext/reflection/php_reflection.stub.php
- https://github.com/php/php-src/blob/php-8.0.0/ext/reflection/php_reflection.c

### CODE REVIEW

**Comment by Girgias** [comment]
@beberlei can you review this?

File: reference/reflection/reflectionattribute/isrepeated.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>bool</type><methodname>ReflectionAttribute::isRepeated</methodname>
+   <void />
+  </methodsynopsis>
+  <para>
+   Gets whether attribute is used repeatedly.
```
Comment (beberlei) [inline reference/reflection/reflectionattribute/isrepeated.xml:17]:
```suggestion
   Returns weather the attribute of this name could occur repeatedly on a code element.
```

File: reference/reflection/reflectionattribute/isrepeated.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- $Revision$ -->
+
+<refentry xml:id="reflectionattribute.isrepeated" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
+ <refnamediv>
+  <refname>ReflectionAttribute::isRepeated</refname>
+  <refpurpose>Gets whether attribute is used repeatedly.</refpurpose>
```
Comment (beberlei) [inline reference/reflection/reflectionattribute/isrepeated.xml:7]:
```suggestion
  <refpurpose>Returns weather the attribute of this name could occur repeatedly on a code element.</refpurpose>
```

File: reference/reflection/reflectionattribute/newinstance.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- $Revision$ -->
+
+<refentry xml:id="reflectionattribute.newinstance" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
+ <refnamediv>
+  <refname>ReflectionAttribute::newInstance</refname>
+  <refpurpose>Creates a new ReflectionAttribute instance.</refpurpose>
```
Comment (beberlei) [inline reference/reflection/reflectionattribute/newinstance.xml:7]:
```suggestion
  <refpurpose>Instantiates the attribute class represented by this ReflectionAttribute class and arguments.</refpurpose>
```

File: reference/reflection/reflectionattribute/newinstance.xml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>object</type><methodname>ReflectionAttribute::newInstance</methodname>
+   <void />
+  </methodsynopsis>
+  <para>
+   Creates a new instance of the attribute.
```
Comment (beberlei) [inline reference/reflection/reflectionattribute/newinstance.xml:17]:
```suggestion
   Instantiates the attribute class represented by this ReflectionAttribute class and arguments.
```

File: reference/reflection/reflectionclass/getattributes.xml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>array</type><methodname>ReflectionClass::getAttributes</methodname>
+   <methodparam><type>string</type><parameter>name</parameter><initializer>&null;</initializer></methodparam>
+   <methodparam><type>int</type><parameter>flags</parameter><initializer>0</initializer></methodparam>
+  </methodsynopsis>
+  <para>
+   Get the attributes as an array of <type>ReflectionAttribute</type>.
```
Comment (beberlei) [inline reference/reflection/reflectionclass/getattributes.xml:18]:
```suggestion
   Returns all attributes declared on this class as an array of <type>ReflectionAttribute</type>.
```

File: reference/reflection/reflectionclassconstant/getattributes.xml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>array</type><methodname>ReflectionClassConstant::getAttributes</methodname>
+   <methodparam><type>string</type><parameter>name</parameter><initializer>&null;</initializer></methodparam>
+   <methodparam><type>int</type><parameter>flags</parameter><initializer>0</initializer></methodparam>
+  </methodsynopsis>
+  <para>
+   Get the attributes as an array of <type>ReflectionAttribute</type>.
```
Comment (beberlei) [inline reference/reflection/reflectionclassconstant/getattributes.xml:18]:
```suggestion
   Returns all attributes declared on this class constant as an array of <type>ReflectionAttribute</type>.
```

File: reference/reflection/reflectionfunctionabstract/getattributes.xml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>array</type><methodname>ReflectionFunctionAbstract::getAttributes</methodname>
+   <methodparam><type>string</type><parameter>name</parameter><initializer>&null;</initializer></methodparam>
+   <methodparam><type>int</type><parameter>flags</parameter><initializer>0</initializer></methodparam>
+  </methodsynopsis>
+  <para>
+   Get the attributes as an array of <type>ReflectionAttribute</type>.
```
Comment (beberlei) [inline reference/reflection/reflectionfunctionabstract/getattributes.xml:18]:
```suggestion
   Returns all attributes declared on this function or method as an array of <type>ReflectionAttribute</type>.
```

File: reference/reflection/reflectionparameter/getattributes.xml [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>array</type><methodname>ReflectionParameter::getAttributes</methodname>
+   <methodparam><type>string</type><parameter>name</parameter><initializer>&null;</initializer></methodparam>
+   <methodparam><type>int</type><parameter>flags</parameter><initializer>0</initializer></methodparam>
+  </methodsynopsis>
+  <para>
+   Get the attributes as an array of <classname>ReflectionAttribute</classname>.
```
Comment (beberlei) [inline reference/reflection/reflectionparameter/getattributes.xml:18]:
```suggestion
   Returns all attributes declared on this parameter as an array of <type>ReflectionAttribute</type>.
```

File: reference/reflection/reflectionproperty/getattributes.xml [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>array</type><methodname>ReflectionProperty::getAttributes</methodname>
+   <methodparam><type>string</type><parameter>name</parameter><initializer>&null;</initializer></methodparam>
+   <methodparam><type>int</type><parameter>flags</parameter><initializer>0</initializer></methodparam>
+  </methodsynopsis>
+  <para>
+   Get the attributes as an array of <classname>ReflectionAttribute</classname>.
```
Comment (beberlei) [inline reference/reflection/reflectionproperty/getattributes.xml:18]:
```suggestion
   Returns all attributes declared on this class property as an array of <type>ReflectionAttribute</type>.
```

**Review by beberlei — CHANGES_REQUESTED** [review]
This is really great work, thank you so very much! I have added just a few comments on better descriptions for methods and then this is good to ship from my POV.

**Comment by mumumu** [comment]
@beberlei Thank you very much! I modified this PR in accordance with your suggestions.

**Review by cmb69 — APPROVED** [review]
Thank you!

While I don't like to see `&warn.undocumented.func;` in the manual, it's better than no documentation at all. :)

**Comment by mumumu** [comment]
> While I don't like to see &warn.undocumented.func; in the manual, it's better than no documentation at all. :)

Yeah, I'll resolve `&warn.undocumented.func;` issue on #659.

**Pull request merged**